internal/guest/runtime/runc/container.go

```diff
@@ -366,6 +370,26 @@ func (c *container) startProcess(
 	cmd := runcCommandLog(logPath, args...)
 
 	var pipeRelay *stdio.PipeRelay
+	var fifoPipe *os.File
+
+	pipeName, exists := annotations["io.microsoft.bmc.logging.pipelocation"]
+	if exists {
+		_, err = os.Stat(pipeName)
+		if err != nil {
+			// fifo pipe does not exist, create one
+			err = syscall.Mkfifo(pipeName, 0666)
```

Comment:
this is ok.. in case 2 calls tried to make the same pipe.. one of them will be no op.

Reply:
There is still a [TOCTTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) issue here since we first check if the pipe exists (`os.Stat`) and then create it (`syscall.Mkfifo`). If there are two concurrent container creates, we could still hit a case here where they both call `syscall.Mkfifo`.

That said, given this is an edge case, if hit it will simply result in a container creation failure, and that we will be changing this code later, I'm okay keeping this as-is for now.